spanner/ddl/ast_test.go

```diff
@@ -101,21 +102,31 @@ func TestPrintCreateTable(t *testing.T) {
 }
 
 func TestPrintCreateIndex(t *testing.T) {
-	ci := CreateIndex{
-		"myindex",
-		"mytable",
-		[]IndexKey{IndexKey{Col: "col1", Desc: true}, IndexKey{Col: "col2"}},
-	}
+	ci := []CreateIndex{
+		CreateIndex{
+			"myindex",
+			"mytable",
+			/*Unique =*/ false,
+			[]IndexKey{IndexKey{Col: "col1", Desc: true}, IndexKey{Col: "col2"}},
+		},
+		CreateIndex{
+			"myindex2",
+			"mytable",
+			/*Unique =*/ true,
+			[]IndexKey{IndexKey{Col: "col1", Desc: true}, IndexKey{Col: "col2"}},
+		}}
 	tests := []struct {
 		name       string
 		protectIds bool
+		index      CreateIndex
 		expected   string
 	}{
-		{"no quote", false, "CREATE INDEX myindex ON mytable (col1 DESC, col2)"},
-		{"quote", true, "CREATE INDEX `myindex` ON `mytable` (`col1` DESC, `col2`)"},
+		{"no quote non unique", false, ci[0], "CREATE INDEX myindex ON mytable (col1 DESC, col2)"},
+		{"quote non unique", true, ci[0], "CREATE INDEX `myindex` ON `mytable` (`col1` DESC, `col2`)"},
+		{"unique key", true, ci[1], "CREATE UNIQUE INDEX `myindex2` ON `mytable` (`col1` DESC, `col2`)"},
 	}
 	for _, tc := range tests {
-		assert.Equal(t, normalizeSpace(tc.expected), normalizeSpace(ci.PrintCreateIndex(Config{ProtectIds: tc.protectIds})))
+		assert.Equal(t, normalizeSpace(tc.expected), normalizeSpace(tc.index.PrintCreateIndex(Config{ProtectIds: tc.protectIds})))
```

Comment:
[optional] while we are here, let's get rid of normalizeSpace if possible - if it turns to be rather a lot of work, then let's leave it as it is since this is existing code

Reply:
Yeah, better to handle that in separate PR

Reply:
ok, sounds good